### PR TITLE
refractor course upload

### PR DESCRIPTION
**Pull request by j2kan**
1. Got rid of temp course table. Got rid of 3 html pages related to this functionality to 1.
2. Made upload more robust and error messages for everything I can think of.
3. Verified Dr. Smucker's original Spring CSV with extra columns and rows and missing field names still upload
4. Started to migrate code that's not related to request context into utils.
5. Fixed the 2 unit test we had for student and application form

### CODE REVIEW

**Comment by smwatts** [comment]
*add wording saying that you can continue to upload courses

Need to fix the pop-up location

screen shot 2018-03-11 at 2 11 53 pm

**Comment by smwatts** [comment]
[course_template-16.xlsx](https://github.com/j2kan/ta-works/files/1800705/course_template-16.xlsx)

Result:
screen shot 2018-03-11 at 2 20 45 pm

**Review by lauradobson — CHANGES_REQUESTED** [review]
Good error checking:

Following changes:
1. Better displays of error messages
2. Center pop up
3. Is it possible to give the row number in the excel file when this is the error message? "The following row needs term to be an integer: [' ', ' ', ' ', ' ', ' ', ' ', ' ']"

**Comment by amyleblond** [comment]
screen shot 2018-03-11 at 2 23 42 pm
on smucker file upload ^

**Comment by lauradobson** [comment]
screen shot 2018-03-11 at 2 25 24 pm

on smuckers spring 2018 file

**Comment by amyleblond** [comment]
remove me please "-"

screen shot 2018-03-11 at 2 30 53 pm

**Comment by lauradobson** [comment]
save as "CSV - UTF 8" doesn't work but save as "CSV" works

File: taworks/README.md [hunk 1]
```diff
@@ -52,9 +52,14 @@ _after running the queries below, you will need to press `q` to get out of it_
     * verify you can only land on this page if you're logged in
     * change the password back to the original password for 'taform'
     * logout and login with original password
-4. [Upload Courses](https://team4.uwaterloo.ca/taform/taform/course_list.html)
-    * verify CSV template can be downloaded and download it
-    * upload a subset of this downloaded template
+4. [Upload Courses](https://team4.uwaterloo.ca/taform/taform/upload_course_list.html)
```
Comment (lauradobson) [inline taworks/README.md:55]: Add the CSV extension issue here as well

File: taworks/taform/utils/upload.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    )
+    course.save()
+
+def check_field_name(errors, fieldnames):
+    expected = ['term', 'course_subject', 'course_id', 'section', 'course_name', 'instructor_name', 'instructor_email']
+    if fieldnames:
+        for i in expected:
+            if i not in fieldnames:
+                errors.append("The following column header is missing: " + i)
+        errors.append("Make sure your CSV file is ',' delimited and not tab or space delimited.")
```
Comment (lauradobson) [inline taworks/taform/utils/upload.py]: Can this line go to an else for this if statement?

**Review by lauradobson — APPROVED** [review]
1 comment otherwise lgtm

**Review by amyleblond — APPROVED** [review]

**Comment by lauradobson** [comment]
push to one line
screen shot 2018-03-11 at 3 56 44 pm

**Review by amyleblond — APPROVED** [review]

**Pull request merged**